src/openqaoa-core/utilities.py

```diff
@@ -1427,6 +1488,76 @@ def flip_counts(counts_dictionary: dict) -> dict:
     return output_counts_dictionary
 
 
+def permute_counts_dictionary(
+    counts_dictionary: dict, final_qubit_layout: List[int]
+) -> dict:
+    """Permutes the order of the qubits in the counts dictionary to the
+    original order if SWAP gates were used leading to modified qubit layout.
+    Parameters
+    ----------
+    counts_dictionary : `dict`
+        The measurement outcomes obtained from the Simulator/QPU
+    original_qubit_layout: List[int]
+        The qubit layout in which the qubits were initially
+    final_qubit_layout: List[int]
+        The final qubit layout after application of SWAPs
+
+    Returns
+    -------
+    `dict`
+        The permuted counts dictionary with qubits in the original place
+    """
+
+    # Create a mapping of original positions to final positions
+    original_qubit_layout = list(range(len(final_qubit_layout)))
+    mapping = {
+        original_qubit_layout[i]: final_qubit_layout[i]
+        for i in range(len(original_qubit_layout))
+    }
+    permuted_counts = {}
+
+    for basis, counts in counts_dictionary.items():
+
+        def permute_string(basis_state: str = basis, mapping: dict = mapping):
+            # Use the mapping to permute the string
+            permuted_string = "".join(
+                [basis_state[mapping[i]] for i in range(len(basis_state))]
+            )
+            return permuted_string
+
+        permuted_counts.update({permuted_string: counts})
+
+    return permuted_counts
+
+
+def negate_counts_dictionary(counts_dictionary: dict, s: int) -> dict:
```

Comment:
Is the negation just a bitflip? If so maybe bitflip is easier to read than negate?